compiler/sigmatch.nim

```diff
@@ -2432,7 +2432,7 @@ proc matchesAux(c: PContext, n, nOrig: PNode, m: var TCandidate, marker: var Int
     return
 
   template checkConstraint(n: untyped) {.dirty.} =
-    if not formal.constraint.isNil:
+    if not formal.constraint.isNil and not (formal.constraint.kind == nkStrLit and "$#" in formal.constraint.strVal):
```

Comment:
No. Too hacky.

Reply:
I thought it but didnt want to introduce "yet another flag", I assumed you had your reason when you did/accepted `codegenDecl` in the first place. 
But rethinking it now, maybe we could have `sfCodegenDecl` and that would add clarity to all checks over the place? (right now they mostly do `constraint != nil`)

Reply:
`sfCodegenDecl` is better.